### PR TITLE
Bug 1856708 - Fix ReaderView to use `storage.session` correctly

### PR DESCRIPTION
**Pull request by willdurand**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1856708

### CODE REVIEW

File: android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js [hunk 1]
```diff
@@ -17,11 +17,11 @@ browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
        });
        break;
      case 'addSerializedDoc':
-        storage.session.set(sender.contextId.toString(), message.doc);
+        browser.storage.session.set(sender.contextId.toString(), message.doc);
```
Comment (zmckenney) [inline android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js:20]: The arguments for `browser.storage.session.set` are incorrect and I think we should be waiting (?) so suggesting an alternative. 

Also, in debugging I found that `addSerializedDoc` is called twice and `getSerializedDoc` is called twice when opening a readerview. After each closing of readerview when the user goes to open the same page in readerview the `contextId` has changed and it again calls `addSerializedDoc` and `getSerializedDoc` twice.

Unless I'm understanding incorrectly, shouldn't the correct behavior be the first opening of the readerview we add the serialized doc with a persistent contextId (or another type of UID) and then subsequent openings of this page call we just call `getSerializedDoc` and return the cached doc?
```suggestion
        let sessionContextKey = sender.contextId.toString();
        await browser.storage.session.set({sessionContextKey: message.doc});
```

**Comment by boek** [comment]
@willdurand Possible I wasn't doing it right but it appears that `.set` requires an object now not a key and value parameter. I also took a whack at fixing this with https://github.com/mozilla-mobile/firefox-android/pull/3922

**Review by zmckenney — CHANGES_REQUESTED** [review]

**Comment by willdurand** [comment]
> @willdurand Possible I wasn't doing it right but it appears that `.set` requires an object now not a key and value parameter. I also took a whack at fixing this with https://github.com/mozilla-mobile/firefox-android/pull/3922

sigh, right.. can that patch be uplifted to beta?

File: android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js (hunk 1 above)
Comment (willdurand) [reply android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js:20]: > The arguments for `browser.storage.session.set` are incorrect and I think we should be waiting (?) so suggesting an alternative.

right...

> the `contextId` has changed

looks like a pre-existing issue, but yeah...

**Comment by willdurand** [comment]
Let's close this in favor of https://github.com/mozilla-mobile/firefox-android/pull/3922

**Pull request closed without merging**